src/promptflow-devkit/promptflow/_sdk/operations/_flow_operations.py

```diff
@@ -1032,43 +1033,85 @@ def _infer_signature(self, entry: Callable, keep_entry: bool = False, validate:
             del flow_meta["entry"]
         return flow_meta, source_path.parent
 
+    @monitor_operation(activity_name="pf.flows.infer_signature", activity_type=ActivityType.PUBLICAPI)
+    def infer_signature(self, entry: Callable) -> dict:
+        """Extract signature for a callable class or a function. Signature indicates the ports of a flex flow using
+        the callable as entry.
+
+        If entry is a callable function, the signature includes inputs and outputs. For example:
+        def my_func(a: int, b: str) -> int:
+            pass
+        Signature:
+        {
+            "inputs": {
+                "a": {
+                    "type": "int",
+                },
+                "b": {
+                    "type": "string",
+                },
+            },
+            "outputs": {
+                "output": {
+                    "type": "int",
+                },
+            },
+        }
+
+        If entry is a callable class, the signature includes inputs, outputs, and init. For example:
+        @dataclasses.dataclass
```

Comment:
Let's give TypeDict as the default sample

Reply:
It will involve changes in tool generator, let's do this in a separate PR